cetmix_tower_server/tests/test_variable.py

```diff
@@ -659,3 +662,63 @@ def test_get_by_variable_reference(self):
         self.assertEqual(
             server_result.get("server_template"), server_template_value.value_char
         )
+
+    def test_single_assignment(self):
+        """Test that a variable can only be assigned to one model at a time."""
+        # Create a variable value assigned to the server
+        variable_value = self.env["cx.tower.variable.value"].create(
+            {
+                "variable_id": self.variable_os.id,
+                "value_char": "Branch = Main",
+                "server_id": self.server_test_1.id,
+            }
+        )
+
+        # Try to assign the same variable value to
+        # server template and expect a ValidationError
+        with self.assertRaises(ValidationError):
+            variable_value.write({"server_template_id": self.server_template_sample.id})
+
+        # Try to assign the same variable value to
+        # plan line action and expect a ValidationError
+        with self.assertRaises(ValidationError):
+            variable_value.write({"plan_line_action_id": self.plan_line_1_action_1.id})
+
+    def test_unique_assignment(self):
+        """Test that the same variable value cannot be
+        assigned multiple times to the same record.
+        """
+
+        variable = self.env["cx.tower.variable"].create(
+            {"name": "Test Variable Custom", "note": "A variable for testing."}
+        )
+        server = self.env["cx.tower.server"].create(
+            {
+                "name": "Test 2",
+                "ip_v4_address": "localhost1",
+                "ssh_username": "admin1",
+                "ssh_password": "password1",
+                "ssh_auth_mode": "p",
+            }
+        )
+
+        self.env["cx.tower.variable.value"].create(
+            {
+                "variable_id": variable.id,
+                "value_char": "Branch = Main",
```

Comment:
This is very strange value although correct.
Final expression will look like `Test Variable Custom = Branch = Main`
So you should find some more relevant value here. Even if this is just a test)

Reply:
Thanks for review, I'll fix it